maven/impl-maven-embedded/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/MavenDownloadsTestCase.java

```diff
@@ -56,6 +65,60 @@ public void testDownloadFromGivenDestinationWithoutUsingCacheAndExtract() throws
         verifyExtraction(2, "resolver-3.0.0-alpha-1", "resolver-3.0.0-alpha-3");
     }
 
+    @Test
+    public void testDownloadInMultipleThreads() throws IOException, InterruptedException {
+        // cleanup
+        FileUtils.deleteDirectory(targetMavenDir);
+
+        // multiple download
+        CountDownLatch latch = new CountDownLatch(1);
```

Comment:
yeah, two latches or wait until both threads are stopped. https://github.com/shrinkwrap/resolver/pull/120/files/88f2b2f8b038a8facec04866da2b4a574367d2bd#diff-9df706518aa71a393efa82332a01b03aR83
two latches would be probably less of code...